app/src/main/java/net/gsantner/markor/format/ActionButtonBase.java

```diff
@@ -474,36 +474,48 @@ public static void runRegexReplaceAction(final Editable editable, final ReplaceP
 
     private static void runRegexReplaceAction(final Editable editable, final List<ReplacePattern> patterns) {
 
-        TextViewUtils.withKeepSelection(editable, (selStart, selEnd) -> {
+        final int[] sel = TextViewUtils.getSelection(editable);
+        if (sel[0] < 0) {
+            return;
```

Comment:
sel[0] < 0 means invalid / no selection. 

We were not testing for this which means actions were likely silently failing. Added these tests